### ISSUE TITLE
Please provide vendor tarballs for a release

### ISSUE THREAD

**Issue by Schievel1** [lapce/lapce#2247]
## Is your feature request related to a problem? Please describe.
tl;dr: Modern languages make packagers problems. There is an easy solution, look below!

Long version:
Most package systems are not build to fit modern statically linked languages that have their own package managers that manage dependencies. Most of them come from an era where developers released tarballs an their own website. A packager would write a package that fetches this tarball, set some options and compiled it. 
With the modern languages package managers adapted by adding possibilities to fetch those dependencies from the internet, doing Git checkouts instead of downloading a tarball etc. 
However, how those languages are handled in package managers is still open to debate. 
For example there are package managers that just fetch the source and run `cargo build`, some package each dependency independently and manage everything their own. For Rust packages on Gentoo we decided that we utilize Cargo, but we do not allow cargo to fetch anything from the internet during build.
We therefore have a variable with a list of crates. All the crates on that list are fetched and put into the right place, then we patch the cargo config of the package and only then invoke cargo build (with --offline)

This mechanism worked nice for a while until we ran into problems with the same mechanism we use with packages written in Go-Lang. For some packages that list (which is just a space-separated string) was getting longer than what the Linux Kernel would allow for a variable. (https://bugs.gentoo.org/721088) Also for every artifact that is downloaded during build we check the checksum. The checksum is saved during packaging. Those lists with checksums also were getting too big for Go-Lang packages. 
Therefore the deprecated that mechanism for Go-Lang packages. On Rust it is not that bad, and we kept it for now, but for packagers it is still quite much work to figure out the dependency crates and especially with crates from Github that lapce uses heavily. Because we need to get the commit hash of every crate from Github that lapce uses every time a new version of lapce is released. 

## Describe the solution you'd like
I would like to ask you to release a vendor tarball alongside ever release. A vendor tarball is a tar archive that contains all of the dependency crates. 
It is easily generated by running `cargo vendor` in the repository and then tarring that created folder with `tar -czvf vendor.tar.gz vendor`.
The resulting tarball could be released as a release artifact. 

If you are using GitHub actions to manage your releases, this could be easily done by running a few extra jobs of checking out the repo, running cargo vendor and tarring that vendor directory. 

This is just a few extra lines in a GitHub job for you, but it would help me tremendously to package your software and make it easily available for Gentoo users.

**Comment by Schievel1** [comment]
Also, to get an idea of what I am talking about, here is the ebuild for lapce-0.2.6. https://github.com/gentoo/guru/blob/c680d375d70dc13d0e055f944014711735f3e67c/app-editors/lapce/lapce-0.2.6.ebuild#L575
An ebuild is essentially a skript on how to build a package. On gentoo, as users build the software themselves on their computers, an ebuild IS the package. 

With every lapce release I need to regenerate that list (we have tools to automate that) and the list with the GIT_CRATES with their new commits. (by hand by looking in the Cargo.toml and Cargo.lock files and copying the commits over to the ebuild)

**Comment by panekj** [comment]
Fixed via #2709

**Comment by Schievel1** [comment]
Thanks for that. I just bumped lapce to 0.3.0 on Gentoo using the vendor tarball. Works like a charm.

**Comment by Schievel1** [comment]
I just bumped the Gentoo's lapce package to 0.3.1 for Gentoo and it went absolutely smooth. In fact I just changed the file name of the ebuild file. 

I just wanted to thank you again, this makes things so much faster an easier.

**Comment by Schievel1** [comment]
Hi, 

could you add the vendor tar for 0.4.3 again?

**Comment by F3llFr0mTh3Sky** [comment]
There's no vendor for 0.4.4 and 0.4.5 either. Could you please publish them?
